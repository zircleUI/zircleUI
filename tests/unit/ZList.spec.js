import { shallowMount, createLocalVue } from '@vue/test-utils'
import zlist from '@/components/z-list.vue'
import zlistpagination from '@/components/child-components/z-pagination.vue'
import zlistitem from '@/components/z-spot.vue'
import zircle from '@/index'

const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zlist, {
  propsData: {
    items: [{ name: 'Apple', color: 'red' }, { name: 'Apple1', color: 'green' }],
    perPage: 5
  },
  provide: function () {
    return {
      view: 'test'
    }
  },
  computed: {
    position: function () {
      return { X: 0, Y: 0, scale: 1, Xi: 0, Yi: 0, scalei: 1 }
    }
  },
  // Currently vue-test-utils is working in the implementation of scopedSlots.
  // scopedSlots: {
  // props: '<z-item slot-scope="props" :index="props.index" :color="props.color" :to-view="/item/' + props.name + '"> <h1 slot="label"> {{props.index}}</h1></z-item>'
  // },
  stubs: {
    'z-pagination': zlistpagination,
    'z-spot': zlistitem
  },
  localVue
})
describe('z-list.vue', () => {
  it('Renders props when passed', () => {
    expect(wrapper.vm.items).toEqual([{ name: 'Apple', color: 'red' }, { name: 'Apple1', color: 'green' }])
    expect(wrapper.vm.perPage).toEqual(5)
  })
  it('Has the expected html structure', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
