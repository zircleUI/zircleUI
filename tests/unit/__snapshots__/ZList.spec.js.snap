// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-list.vue Has the expected html structure 1`] = `
<section
  class="z-list"
>
  <div />
  <div />
   
  <!---->
</section>
`;
